src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/TemplateBuilder.cs

```diff
@@ -125,10 +125,17 @@ public IHtmlContent Build()
             var formatString = _readOnly ?
                 viewData.ModelMetadata.DisplayFormatString :
                 viewData.ModelMetadata.EditFormatString;
+
             if (_model != null && !string.IsNullOrEmpty(formatString))
             {
                 formattedModelValue = string.Format(CultureInfo.CurrentCulture, formatString, formattedModelValue);
             }
+            else if (string.Equals("week", _templateName, StringComparison.OrdinalIgnoreCase) ||
+                string.Equals("week", viewData.ModelMetadata.DataTypeName, StringComparison.OrdinalIgnoreCase))
+            {
+                // "week" is a new HTML5 input type that only will be rendered in Rfc3339 mode
+                formattedModelValue = FormatWeekHelper.GetFormattedWeek(_modelExplorer);
```

Comment:
This will be called when `_modelExplorer.Model` is `null`. Make sure that doesn't happen in the new location.